### ISSUE TITLE
[Java][5.5.0.2] Actor interface methods with same name collide

### ISSUE THREAD

**Issue by erikogenvik** [Azure/service-fabric-issues#275]
I'm running Service Fabric for Java 5.5.0.2.

The issue is that if you have two Actor interfaces where both of them have a method named the same way, you'll get a java.lang.LinkageError when calling the "second" one through an ActorProxyBase.

I.e. when the methods are called through ActorProxyBase the code will do some dynamic class creation (the first time the actor proxy is created). It seems that the generated classes all share the same "namespace", so that when the second invokation happens the class needed for handling the request body ("*ReqBody") of the duplicate method already is created. Here's an edited stack dump of this error (with the duplicate method "fooBar"):
```
 java.lang.LinkageError: loader (instance of  microsoft/servicefabric/services/remoting/builder/CodeBuilderClassLoader): attempted  duplicate class definition for name: "fooBarReqBody"
         at java.lang.ClassLoader.defineClass1(Native Method)
         at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
         at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
         at microsoft.servicefabric.services.remoting.builder.CodeBuilderClassLoader.defineClass(CodeBuilderClassLoader.java:31)
         at microsoft.servicefabric.services.remoting.builder.CodeBuilderContext.complete(CodeBuilderContext.java:29)
         at microsoft.servicefabric.services.remoting.builder.MethodBodyTypesBuilder.buildRequestBodyType(MethodBodyTypesBuilder.java:93)
         at microsoft.servicefabric.services.remoting.builder.MethodBodyTypesBuilder.build(MethodBodyTypesBuilder.java:54)
         at microsoft.servicefabric.services.remoting.builder.MethodBodyTypesBuilder.build(MethodBodyTypesBuilder.java:38)
         at microsoft.servicefabric.actors.ActorCodeBuilder.buildMethodBodyTypes(ActorCodeBuilder.java:60)
         at microsoft.servicefabric.services.remoting.builder.CodeBuilderBase.getOrBuildMethodBodyTypes(CodeBuilderBase.java:39)
         at microsoft.servicefabric.services.remoting.builder.ProxyGeneratorBuilder.build(ProxyGeneratorBuilder.java:49)
         at microsoft.servicefabric.actors.ActorCodeBuilder.buildProxyGenerator(ActorCodeBuilder.java:86)
         at microsoft.servicefabric.services.remoting.builder.CodeBuilderBase.getOrBuildProxyGenerator(CodeBuilderBase.java:49)
         at microsoft.servicefabric.actors.ActorCodeBuilder.getOrCreateProxyGenerator(ActorCodeBuilder.java:35)
         at microsoft.servicefabric.actors.FabricActorProxyFactory.createActorProxy(FabricActorProxyFactory.java:90)
         at microsoft.servicefabric.actors.ActorProxyBase.create(ActorProxyBase.java:253)
         at microsoft.servicefabric.actors.ActorProxyBase.create(ActorProxyBase.java:222)
         ...
```
Also note that this throws an Error rather than an Exception.

**Comment by PavanKunapareddyMSFT** [comment]
Thanks Erik for reporting the issue.
This is a known bug. We have fixed this and will be available in the upcoming release.

**Comment by raunakpandya** [comment]
Update: The fix for this would be part of the next SDK update on Linux which is expected to be out within a week.

**Comment by raunakpandya** [comment]
This has been fixed in the latest Java SDK v 0.9.0.0
